packages/manager/apps/cloud/client/app/dbaas/logs/detail/streams/home/logs-streams-home.controller.js

```diff
@@ -117,13 +138,40 @@ class LogsStreamsHomeCtrl {
     });
   }
 
+  loadRetentions() {
+    forEach(this.accountDetails.data.clusters, (cluster) => {
+      forEach(cluster.retentions, (retention) => {
+        this.retentions.push(retention);
+      });
+    });
+  }
+
   storageInfo(stream) {
     if (stream.info.isEditable && stream.info.currentStorage !== -1) {
       return this.bytesFilter(stream.info.currentStorage, 2, true);
     }
     return ' - ';
   }
 
+  findRetention(stream) {
+    return find(
+      this.retentions,
+      (retention) => retention.retentionId === stream.info.retentionId,
+    );
+  }
+
+  retentionInfo(stream) {
+    const foundRetention = this.findRetention(stream);
+    if (foundRetention.duration) {
+      if (foundRetention.duration === 'P45D') {
+        // moment convert 45 days to a month and we need to be accurate here
+        return this.$translate.instant('streams_45_days');
```

Comment:
Nice to know why it is done that way :) I don't know if this could be achieved by using `humanize` threshold parameter (https://momentjs.com/docs/#/durations/humanize/)